src/HTTPCommands.cc

```diff
@@ -130,7 +130,7 @@ bool HTTPRequest::parseProtocol(const std::string &url, std::string &protocol) {
 	return true;
 }
 
-bool HTTPRequest::SendHTTPRequest(const std::string &payload) {
+bool HTTPRequest::SendHTTPRequest(const std::string &payload, bool final) {
```

Comment:
Unclear if the `final` is needed since libcurl knows how large the request should be.  Please give it a try without now that we've got more experience with the branch.